tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -934,3 +896,38 @@ async fn maybe_fetch_cache_info(
         maybe_slot_leaders,
     }
 }
+
+/// Create LeaderTpuService
+/// Retries until successful or timeout is reached
+async fn create_leader_tpu_service<P, M, C>(
+    rpc_client: Arc<RpcClient>,
+    websocket_url: &str,
+    exit: Arc<AtomicBool>,
+) -> Result<LeaderTpuService>
+where
+    P: ConnectionPool<NewConnectionConfig = C>,
+    M: ConnectionManager<ConnectionPool = P, NewConnectionConfig = C>,
+    C: NewConnectionConfig,
```

Comment:
I do not think that any of `P`, `M` or `C` are used in the function implementation, except for `M::PROTOCOL`.
And `M::PROTOCOL` is actually a constant, not a generic parameter.
It is a value of type `solana_connection_cache::connection_cache::Protocol`.
Meaning, this function does not need to be generic at all.